indra/newview/llappviewer.cpp

```diff
@@ -2171,7 +2171,12 @@ bool LLAppViewer::initThreads()
 
     // get the number of concurrent threads that can run
     S32 cores = std::thread::hardware_concurrency();
-
+#if LL_DARWIN
+    if (!gGLManager.mIsApple)
+    {
+        cores /= 2;
```

Comment:
is this an SMT/hyperthreading workaround?  is there a way to detect that?

Reply:
Pretty much, yeah... I don't know of a way to detect that... on Windows we don't seem to get thread starvation as much when treating virtual cores as real cores, and all seems fine on Apple silicon.  Note that the viewer process still gets blocked pretty hard sometimes (all threads pause), but with that change I don't see what looks like thread starvation when decoding textures.